File: src/leetcode/python/PerfectNumber_507.py
class PerfectNumber_507(object):
    def checkPerfectNumber(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num <= 2:
            return False

        end = int(num ** 0.5)
        res = sum(i + num // i for i in range(2, end + 1) if not num % i)
        res += 1
        return res == num

File: src/leetcode/java/PerfectNumber_507.java
package leetcode.java;

public class PerfectNumber_507 {
  public boolean checkPerfectNumber(int num) {
    if (num == 1) return false;
    int end = (int) Math.sqrt(num);
    int sum = 0;
    for (int start = 2; start <= end; start++) {
      if (num % start == 0) {
        sum += start;
        sum += num / start;
      }
    }
    return ++sum == num;
  }
}
